### ISSUE TITLE
Follow-up to #5999: Fixed a CSS regression with the `#indentation` FAPI property.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6090]
I was working on a quick mockup for the feedback/suggestions I have provided in https://github.com/backdrop/backdrop-issues/issues/2370#issuecomment-1529008603. When I tried adding `'#indentation' => 1` to the radios that are being introduced as part of that feature request, I noticed that the form breaks:

![image](https://user-images.githubusercontent.com/2423362/235352649-ff5baaa6-7217-4c3c-9f4b-ff7a65754c70.png)

I've ruled out the modal being the culprit, because if you open the block configuration form in another browser tab (right-click on the "CONFIGURE" button instead of a regular click), the form is still broken in the exact same manner:
![image](https://user-images.githubusercontent.com/2423362/235352730-9fe08be3-79f9-4ac5-82b3-e1254701db69.png)

**Comment by argiepiano** [comment]
Thanks, @klonos. For the sake of completeness, can you add steps to reproduce here, rather than referring to a separate issue? I haven't followed that issue, and I'm finding hard to understand how this happened.

**Comment by klonos** [comment]
Sure @argiepiano ...I only meant to quickly raise this issue as a placeholder for now (hence the "TBD" in the page title), so that I don't forget about it. The way I got what you see in the screenshots above was to simply add `'#indentation' => 1` to a set of radios and a checkbox in the PR for #2370.

Please see the following (commented-out) lines in the diff between @laryn's branch and mine:
- [laryn/backdrop@`1.x-issue-2370...klonos:backdrop:issue-2370`#diff-f4cf8ec9f0](https://github.com/laryn/backdrop/compare/1.x-issue-2370...klonos:backdrop:issue-2370#diff-f4cf8ec9f05c38c938bd69b000def85260bafed5cf746c0cbd71ac09cd57bb58R159)
- [laryn/backdrop@`1.x-issue-2370...klonos:backdrop:issue-2370`#diff-f4cf8ec9f0](https://github.com/laryn/backdrop/compare/1.x-issue-2370...klonos:backdrop:issue-2370#diff-f4cf8ec9f05c38c938bd69b000def85260bafed5cf746c0cbd71ac09cd57bb58R176)

**Comment by klonos** [comment]
Confirmed that this is broken elsewhere. I see things broken in the image field settings form while working on #3006 and after I rebased with recent changes in 1.x:
![image](https://user-images.githubusercontent.com/2423362/235486318-b0584967-aaaa-468b-97ab-5f0484b19944.png)

**Comment by argiepiano** [comment]
I'm investigating this problem. FTR, I tried removing the comment for #indentation in #3006 using a dev version I downloaded a couple of weeks ago, and the indentation works as expected. So, this is a regression that was created after #6028 was merged.

**Comment by argiepiano** [comment]
I found it! https://github.com/backdrop/backdrop-issues/issues/6028 is the culprit. It introduced css class `indentation` (not sure why). That class existed in Backdrop for a different purpose, and it was not part of the PR that added `#indentation`. #6028 was merged without sufficient testing...

PR https://github.com/backdrop/backdrop/pull/4425

**Comment by klonos** [comment]
@argiepiano sorry for missing that 😞  ...the original implementation of that PR was with an `indent` class, which wasn't conflicting with anything else - it was changed to `indentation` with [backdrop/backdrop@`c275dc6` (#4381)](https://github.com/backdrop/backdrop/pull/4381/commits/c275dc628a1083ee94102d6ba56a7f80b3ce2502) after code review feedback, and unfortunately that wasn't tested.

Instead of removing the `indentation` class altogether, can we please replace it with one that is not as generic and which doesn't conflict with other CSS in core? Something like `form-element-indentation` perhaps?

**Comment by klonos** [comment]
...adding the 1.25.0 milestone, since the `#indentation` property is a new feature that's not in 1.24.x.

**Comment by klonos** [comment]
...and can we please also fix the `@todo` in https://github.com/backdrop/backdrop/blob/66bcad62b935bdfb85aa334006063faa6d0a587d/core/modules/image/image.field.inc#L171 as part of this?

**Comment by argiepiano** [comment]
> Something like form-item-indentation perhaps? (to much the existing form-item-indentation-* numbered classes)
> ...and can we please also fix the @todo in

Sounds good. PR coming a bit later today

**Comment by klonos** [comment]
Thanks @argiepiano 🙏🏼 ...and I'll then also fix the `@todo`s in my PR for #6093 😉

**Comment by argiepiano** [comment]
@klonos, I just pushed a new commit that adds class `form-item-indentation`. It also de-comments the `indentation` setting in `image_field_instance_settings_form()`.

One point related to the new css class. In my latest commit I chose not to style the `<option>`  elements with `form-item-indentation`. The indentation functionality works in very different ways for  `<option>`. The "pseudo-indentation" is accomplished by adding ASCII characters in front of the string, rather than by adding css styling. Therefore adding such a class to `<option>` may lead to unpredictable and unintended consequences. In addition, [according to this article](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option#styling_with_css), it's pretty impossible to apply any type of css styling to `option` - it's still very browser-dependent.   In my opinion, they should not have css classes, but I look forward to hearing yours, of course.

**Comment by klonos** [comment]
Thanks @argiepiano 🙏🏼 ...I have confirmed that this has fixed the issue with the image fields configuration form, and I didn't notice anything else obvious being broken. Code also LGTM, so this is good to go 👍🏼 

Pinging @quicksketch since this is blocking a couple more PRs where we had to comment-out the `'#indentation'` property because of the regression.

**Comment by quicksketch** [comment]
Looks great! I merged https://github.com/backdrop/backdrop/pull/4425 into 1.x. Thanks @argiepiano and @klonos!
